main.js

```diff
@@ -0,0 +1,102 @@
+// submit post when button (id = submit) clicked
+$('#submit').click(function () {
+  var postName = $("#name-input").val();
+  var postMsg = $("#post-input").val();
+
+  // make sure msg or name aren't blank
+  if (postName === '') {
+    alert('Please include your name');
+  };
+
+  if (postMsg === '') {
+    alert('Please post something first');
+  }
+
+  if (postName != '' && postMsg != '') {
+    var $newHTML =
+      `
+      <div id="post1">
+        <span id="remove-post">remove </span><span id="show-or-hide-comments">comments </span>` +
+      postMsg +
+      ` - Posted By: ` +
+      postName +
+      `<div id="comments">
+              <ul id="comments-list"></ul>
+              <div id="comments-list-form"></div>
+            </div>
+            <hr /></div>
+    `;
+
+    $("#posts").append($newHTML);
+
+    // clear inputs
+    $("#name-input").val('');
+    $("#post-input").val('');
+  }
+});
+
+$("#posts").on("click", "#show-or-hide-comments", function () {
```

Comment:
I think this is where things went wrong.

Instead of doing this, add a comments form to every post in the `$('#submit')` callback function and just hide it with CSS, then toggle it visible when it's clicked.

The adding and removing is buggy and too taxing.